src/core/storage/storage.types.ts

```diff
@@ -1,73 +1,40 @@
 import { v4 as uuidv4 } from "uuid";
-import { instanceToPlain, instanceToInstance } from "class-transformer";
 
-export type TagValue = string | boolean | undefined | Array<string> | null;
-export type TagsBase = {
-  [key: string]: TagValue;
-  [key: number]: never;
-};
-
-export type Tags<
-  DefaultTags extends TagsBase,
-  CustomTags extends TagsBase
-> = CustomTags & DefaultTags;
+export type TagsBase = Record<string | number, unknown>;
 
 export type RecordTags<Record extends BaseRecord> = ReturnType<
   Record["getTags"]
 >;
 
-export type BaseRecordAny = BaseRecord<any, any>;
-
-abstract class BaseRecord<
-  DefaultTags extends TagsBase = TagsBase,
-  CustomTags extends TagsBase = TagsBase
-> {
-  protected _tags: CustomTags = {} as CustomTags;
+abstract class BaseRecord {
+  protected _tags: TagsBase = {} as TagsBase;
 
-  public id!: string;
+  id!: string;
 
-  public createdAt!: Date;
+  createdAt!: Date;
 
-  public updatedAt?: Date;
+  updatedAt?: Date;
 
-  public readonly type = BaseRecord.type;
-  public static readonly type: string = "BaseRecord";
+  abstract getTags(): TagsBase;
 
-  public abstract getTags(): Tags<DefaultTags, CustomTags>;
-
-  public setTag(name: keyof CustomTags, value: CustomTags[keyof CustomTags]) {
+  setTag(name: keyof TagsBase, value: TagsBase[keyof TagsBase]) {
     this._tags[name] = value;
   }
 
-  public getTag(name: keyof CustomTags | keyof DefaultTags) {
+  getTag(name: keyof TagsBase | keyof TagsBase) {
     return this.getTags()[name];
   }
 
-  public setTags(tags: Partial<CustomTags>) {
+  setTags(tags: Partial<TagsBase>) {
     this._tags = {
       ...this._tags,
       ...tags,
     };
   }
 
-  public replaceTags(tags: CustomTags & Partial<DefaultTags>) {
+  public replaceTags(tags: TagsBase & Partial<TagsBase>) {
     this._tags = tags;
   }
-
-  public toJSON(): Record<string, unknown> {
-    return instanceToPlain(this, {
-      exposeDefaultValues: true,
-    });
-  }
-
-  public clone() {
-    return instanceToInstance(this, {
-      exposeDefaultValues: true,
-      enableCircularCheck: true,
-      enableImplicitConversion: true,
-      ignoreDecorators: true,
-    });
-  }
 }
 
 type GenericRecordTags = TagsBase;
```

Comment:
Hmm, can we just directly use `TagsBase` and remove this `GenericRecordTags`?

And just call it `Tags`